### PR TITLE
Add instructions on fixing a failed deploy

### PR DESCRIPTION
**Pull request by rtyley**
We  had a bad deploy on Monday, and it exposed that we weren't really sure of the best way to reset the state of the ASG, seems worth documenting!

@philwills @sihil @tackley @chrisjowen @tudorraul

### CODE REVIEW

File: magenta-lib/docs/magenta-lib/howto/fix-a-bad-deploy.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+deploy successfully, and that all live boxes are in the correct state.
+
+
+## A) Manually Kill Bad Boxes
+
+Use the AWS CLI [`terminate-instance-in-auto-scaling-group`](http://docs.aws.amazon.com/cli/latest/reference/autoscaling/terminate-instance-in-auto-scaling-group.html)
+command to kill bad boxes by instance-id, making sure you leave good (old) boxes still operational:
+
+```
+$ aws autoscaling terminate-instance-in-auto-scaling-group --should-decrement-desired-capacity --instance-id i-badbadba
```
Comment (tudorraul) [inline magenta-lib/docs/magenta-lib/howto/fix-a-bad-deploy.md]: I'd put a marker in the command, or a comment, or both underlining that a real instance id should be provided

**Comment by tudorraul** [comment]
:+1: bar the comment

File: magenta-lib/docs/magenta-lib/howto/fix-a-bad-deploy.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+```
+
+Note that you must include `--should-decrement-desired-capacity`, otherwise the ASG will just bring up a new box.
+
+## B) Trust the ASG to Kill the Right Boxes
```
Comment (philwills) [inline magenta-lib/docs/magenta-lib/howto/fix-a-bad-deploy.md]: I'm not convinced it's worth including this.

File: magenta-lib/docs/magenta-lib/howto/fix-a-bad-deploy.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# Fixing a bad deploy
+
+Normally your Auto-Scaling Group will be  configured with a maximum capacity that is _twice_ the 'desired'
+capacity - this is to allow deploys to work by _doubling_ the desired capacity, bringing up new boxes, before
+killing the old ones off.
+
+When a Riff-Raff/Magenta deploy goes bad, your Auto-Scaling Group will often be left in a 'bad' state, where
```
Comment (philwills) [inline magenta-lib/docs/magenta-lib/howto/fix-a-bad-deploy.md]: Happy to merge and then edit further, but I think it's worth explaining the rationale for why RiffRaff adopts this behaviour.

File: magenta-lib/docs/magenta-lib/howto/fix-a-bad-deploy.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
+# Fixing a bad deploy
```
Comment (philwills) [inline magenta-lib/docs/magenta-lib/howto/fix-a-bad-deploy.md]: Can I suggest 'failed', rather than 'bad'.

**Comment by philwills** [comment]
Thanks for doing this @rtyley. I'm happy to merge as is and make the changes myself, if you agree they're sensible.

**Pull request merged**